src/toil/job.py

```diff
@@ -157,11 +159,65 @@ def _parseResource(name, value):
                             % (name, type(value)))
 
 
-class Job(ResourceRequirementMixin):
+class JobLikeObject(ResourceRequirementMixin):
+    def __init__(self, name, job=None, **requirements):
+        super(JobLikeObject, self).__init__(**requirements)
+        self.name = name
+        self.job = job or self.__class__.__name__
+
+    def __str__(self):
+        printedName = '\''+self.job + '\''
+        if self.name:
+            printedName += ' '+self.name
+        return str(printedName)
+
+class JobNode(JobLikeObject):
+    """
+    This object bridges the job graph, job, and batchsystem classes
+    """
+    def __init__(self, memory, cores, disk, preemptable, job, name, jobStoreID,
+                 command, predecessorNumber=1):
+        super(JobNode, self).__init__(memory=memory, cores=cores, disk=disk,
+                                      preemptable=preemptable, name=name, job=job)
+        self.jobStoreID = jobStoreID
+        self.predecessorNumber = predecessorNumber
+        self.command = command
+
+    def __hash__(self):
+        return hash(self.jobStoreID)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        return '%s( **%r )' % (self.__class__.__name__, self.__dict__)
+
+    @classmethod
+    def fromJobGraph(cls, jobGraph):
+        return cls(jobStoreID=jobGraph.jobStoreID, memory=jobGraph.memory,
```

Comment:
In general the rule is to wrap all or none. I'd be ok with grouping all requirements on one line, though, as that would increase readability.